lib/psbt.ex

```diff
@@ -152,6 +192,50 @@ defmodule Bitcoinex.PSBT.Global do
     global = %Global{global | proprietary: value}
     {global, psbt}
   end
+
+  defp serialize_kv(:unsigned_tx, value) when value != nil do
+    key = <<@psbt_global_unsigned_tx::big-size(8)>>
+    key_len = TxUtils.serialize_compact_size_unsigned_int(byte_size(key))
+    val = TxUtils.serialize(value)
+    val_len = TxUtils.serialize_compact_size_unsigned_int(byte_size(val))
+    key_len <> key <> val_len <> val
+  end
+
+  defp serialize_kv(:xpub, value) when value != nil do
+    key = <<@psbt_global_xpub::big-size(8)>>
+    {:ok, key_data} = Base58.decode(value.xpub)
+
+    key = key <> key_data
+
+    key_len = TxUtils.serialize_compact_size_unsigned_int(byte_size(key))
+
+    val =
+      <<value.master_pfp::little-size(32)>> <>
+        (for(chunk <- value.derivation, do: <<chunk::little-size(32)>>)
+         |> :erlang.list_to_binary())
+
+    val_len = TxUtils.serialize_compact_size_unsigned_int(byte_size(val))
+    key_len <> key <> val_len <> val
+  end
+
+  defp serialize_kv(_key, _value) do
```

Comment:
^^ I concur

Reply:
oh true. But then I think we should assume we won't have this impossible state(if our parse PSBT work properly) and therefore no need to pattern match this case and just let it crash in case this happen(maybe user manually construct PSBT struct?)

Reply:
WRT unsigned_tx, I think that is true. I think in the xpub case it should silently ignore it, as a nil xpub field is valid, though I am confused as to why that would be explicitly passed.